pkg/api/routes.go

```diff
@@ -102,8 +106,9 @@ func (rh *RouteHandler) SetupRoutes() {
 			rh.blobLockWrapper(rh.DeleteBlobUpload)).Methods("DELETE")
 		g.HandleFunc("/_catalog",
 			rh.ListRepositories).Methods("GET")
-		g.HandleFunc("/",
-			rh.CheckVersionSupport).Methods("GET")
+		g.HandleFunc("/", rh.CheckVersionSupport).Methods("GET")
+		//nolint (lll)
+		g.HandleFunc("/query", gqlHandler.GraphQL(search.NewExecutableSchema(search.Config{Resolvers: &search.Resolver{DB: rh.c.Config.Extension.Cve.cvedb, Cve: &cveinfo.CveInfo{Log: rh.c.Log}}})))
```

Comment:
also, is this URI path protected with authentication or anyone can access this route?

Reply:
I think it is authenticated since it is a part of same subrouter/route prefix.

Reply:
Pls. cover it with a test case

Reply:
Ok, I will do that